Regression in spacing of left margin in diagnostics

```rust
struct A;
//
//
//
//
//
//
//
fn main() {
    let _: () = A;
    let _: () = B;
}
```

#### `$ cargo +nightly-2019-10-01 check`

This is the correct output.

```console
error[E0425]: cannot find value `B` in this scope
  --> src/main.rs:11:17
   |
11 |     let _: () = B;
   |                 ^ help: a unit struct with a similar name exists: `A`

error[E0308]: mismatched types
  --> src/main.rs:10:17
   |
10 |     let _: () = A;
   |                 ^ expected (), found struct `A`
   |
   = note: expected type `()`
              found type `A`
```

#### `$ cargo +nightly-2019-10-02 check`

Notice that one of the errors has a column of whitespace between the line number and the pipes, the next does not.

```console
error[E0425]: cannot find value `B` in this scope
  --> src/main.rs:11:17
   |
11 |     let _: () = B;
   |                 ^ help: a unit struct with a similar name exists: `A`

error[E0308]: mismatched types
 --> src/main.rs:10:17
  |
10|     let _: () = A;
  |                 ^ expected (), found struct `A`
  |
  = note: expected type `()`
             found type `A`
```

Commit range: https://github.com/rust-lang/rust/compare/22bc9e1d9...702b45e40

It looks like this can cause the wrong line number to be printed. :(

If you bump the problematic line down to line 100, it prints as 10:

```console
error[E0308]: mismatched types
 --> src/main.rs:100:17
  |
10|     let _: () = A;
  |                 ^ expected (), found struct `A`
  |
  = note: expected type `()`
             found type `A`
```

--- a/src/librustc_errors/emitter.rs
+++ b/src/librustc_errors/emitter.rs
@@ -1026,7 +1026,8 @@ impl EmitterWriter {
         children.iter()
             .map(|sub| self.get_multispan_max_line_num(&sub.span))
             .max()
-            .unwrap_or(primary)
+            .unwrap_or(0)
+            .max(primary)
     }
 
     /// Adds a left margin to every line but the first, given a padding length and the label being

